hack/make-rules/test-cmd-util.sh

```diff
@@ -2960,6 +2960,17 @@ runTests() {
   ## test if a short name is exported during discovery
   kube::test::if_has_string "${output_message}" '{"name":"configmaps","singularName":"","namespaced":true,"kind":"ConfigMap","verbs":\["create","delete","deletecollection","get","list","patch","update","watch"\],"shortNames":\["cm"\]}'
 
+  #########################
+  # Assert categories     #
+  #########################
+
+  ## test if a short name is exported during discovery
+	if kube::test::if_supports_resource "${pods}" ; then
+		kube::log::status "Testing propagation of categories for resources"
+		output_message=$(kubectl get --raw=/api/v1)
+		kube::test::if_has_string "${output_message}" '{"name":"pods","singularName":"","namespaced":true,"kind":"Pod","verbs":\["create","delete","deletecollection","get","list","patch","proxy","update","watch"\],"shortNames":\["po"\],"categories":\["all"\]}'
```

Comment:
this is a really long string to match exactly and is likely to drift... can we check just the things we care about?